driver-core/src/test/unit/com/mongodb/internal/operation/BulkWriteBatchSpecification.groovy

```diff
@@ -228,23 +229,71 @@ class BulkWriteBatchSpecification extends Specification {
         !bulkWriteBatch.hasAnotherBatch()
     }
 
-    def 'should only map inserts up to the payload position'() {
+    def 'should map all inserted ids'() {
```

Comment:
I might have made this test more complicated that it needs to be, as I changed it first just to reproduce the bug, but before I settled on the simplified approach in BulkWriteBatch